service/services/services.go

```diff
@@ -0,0 +1,664 @@
+	if tableConf.SortKey != "" {
+		sValue = primaryKeyMap[tableConf.SortKey]
+	}
+	return storage.GetStorageInstance().SpannerGet(ctx, tableName, pValue, sValue, projectionCols)
+}
+
+// QueryAttributes from Spanner
+func QueryAttributes(ctx context.Context, query models.Query) (map[string]interface{}, string, error) {
+	tableConf, err := config.GetTableConf(query.TableName)
+	if err != nil {
+		return nil, "", err
+	}
+	var sKey string
+	var pKey string
+	tPKey := tableConf.PartitionKey
+	tSKey := tableConf.SortKey
+	if query.IndexName != "" {
+		conf := tableConf.Indices[query.IndexName]
+		query.IndexName = strings.Replace(query.IndexName, "-", "_", -1)
+
+		if tableConf.ActualTable != query.TableName {
+			query.TableName = tableConf.ActualTable
+		}
+
+		sKey = conf.SortKey
+		pKey = conf.PartitionKey
+	} else {
+		sKey = tableConf.SortKey
+		pKey = tableConf.PartitionKey
+	}
+	if pKey == "" {
+		pKey = tPKey
+		sKey = tSKey
+	}
+
+	originalLimit := query.Limit
+	query.Limit = originalLimit + 1
+
+	stmt, cols, isCountQuery, offset, hash, err := createSpannerQuery(&query, tPKey, pKey, sKey)
+	if err != nil {
+		return nil, hash, err
+	}
+	logger.LogDebug(stmt)
+	resp, err := storage.GetStorageInstance().ExecuteSpannerQuery(ctx, query.TableName, cols, isCountQuery, stmt)
+	if err != nil {
+		return nil, hash, err
+	}
+	if isCountQuery {
+		return resp[0], hash, nil
+	}
+	finalResp := make(map[string]interface{})
+	length := len(resp)
+	if length == 0 {
+		finalResp["Count"] = 0
+		finalResp["Items"] = []map[string]interface{}{}
+		finalResp["LastEvaluatedKey"] = nil
+		return finalResp, hash, nil
+	}
+	if int64(length) > originalLimit {
+		finalResp["Count"] = length - 1
+		last := resp[length-2]
+		if sKey != "" {
+			finalResp["LastEvaluatedKey"] = map[string]interface{}{"offset": originalLimit + offset, pKey: last[pKey], tPKey: last[tPKey], sKey: last[sKey], tSKey: last[tSKey]}
+		} else {
+			finalResp["LastEvaluatedKey"] = map[string]interface{}{"offset": originalLimit + offset, pKey: last[pKey], tPKey: last[tPKey]}
+		}
+		if query.StartFrom != nil {
+			finalResp["Items"] = resp[:length-1]
+		} else {
+			finalResp["Items"] = resp[:length-1]
+		}
+	} else {
+		if query.StartFrom != nil {
+			if length-1 == 1 {
+				finalResp["Items"] = resp[0]
+			} else {
+				finalResp["Items"] = resp
+			}
+		} else {
+			finalResp["Items"] = resp
+		}
+		finalResp["Count"] = length
+		finalResp["Items"] = resp
+		finalResp["LastEvaluatedKey"] = nil
+	}
+	return finalResp, hash, nil
+}
+
+func createSpannerQuery(query *models.Query, tPkey, pKey, sKey string) (spanner.Statement, []string, bool, int64, string, error) {
+	stmt := spanner.Statement{}
+	cols, colstr, isCountQuery, err := parseSpannerColumns(query, tPkey, pKey, sKey)
+	if err != nil {
+		return stmt, cols, isCountQuery, 0, "", err
+	}
+	tableName := parseSpannerTableName(query)
+	whereCondition, m := parseSpannerCondition(query, pKey, sKey)
+	offsetString, offset := parseOffset(query)
```

Comment:
Is there a reason why some `parse` methods have `Spanner` in the middle and others don't?